file.go

```diff
@@ -171,7 +222,29 @@ func (f *File) SectionStrings() []string {
 }
 
 // DeleteSection deletes a section.
+// If AllowNonUniqueSections is enabled, all sections with the given name are removed.
 func (f *File) DeleteSection(name string) {
+	secs, err := f.GetSections(name)
+	if err != nil {
+		return
+	}
+
+	for i := 0; i < len(secs); i++ {
+		// for non unique sections, always the first one is removed,
+		// so in the next iteration the following section has again index 0
+		// --> just deleting index 0 is enough
+
+		// ignore error as index 0 returns never an error
+		_ = f.DeleteSectionWithIndex(name, 0)
+	}
+}
+
+// DeleteSectionWithIndex deletes a section.
+func (f *File) DeleteSectionWithIndex(name string, index int) error {
+	if !f.options.AllowNonUniqueSections && index != 0 {
+		return fmt.Errorf("error removing section '%s' at index %d. Index greater 0 is only allowed with non unique sections enabled", name, index)
```

Comment:
```suggestion
		return fmt.Errorf("delete section with non-zero index is only allowed when non-unique sections is enabled", name, index)
```

Reply:
Your error string removes the %s and %d. Should I remove the name and index parameters?

Reply:
Yeah, I think removing these two is fine, they're passed as method arguments so the caller knows these info.